internal/provider/resources/resource_team.go

```diff
@@ -0,0 +1,602 @@
+
+	// delete request
+	team, err := r.IamClient.DeleteTeamWithResponse(
+		ctx,
+		r.OrganizationId,
+		data.Id.ValueString(),
+	)
+	if err != nil {
+		tflog.Error(ctx, "failed to delete Team", map[string]interface{}{"error": err})
+		resp.Diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to delete Team, got error: %s", err),
+		)
+		return
+	}
+	statusCode, diagnostic := clients.NormalizeAPIError(ctx, team.HTTPResponse, team.Body)
+	// It is recommended to ignore 404 Resource Not Found errors when deleting a resource
+	if statusCode != http.StatusNotFound && diagnostic != nil {
+		resp.Diagnostics.Append(diagnostic)
+		return
+	}
+
+	tflog.Trace(ctx, fmt.Sprintf("deleted a Team resource: %v", data.Id.ValueString()))
+}
+
+func (r *TeamResource) ImportState(
+	ctx context.Context,
+	req resource.ImportStateRequest,
+	resp *resource.ImportStateResponse,
+) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func (r *TeamResource) ValidateConfig(
+	ctx context.Context,
+	req resource.ValidateConfigRequest,
+	resp *resource.ValidateConfigResponse,
+) {
+	var data models.TeamResource
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Validate if org isScimEnabled
+	org, err := r.PlatformClient.GetOrganizationWithResponse(ctx, r.OrganizationId, nil)
+	if err != nil {
+		tflog.Error(ctx, "failed to validate Team", map[string]interface{}{"error": err})
+		resp.Diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to validate Team, got error: %s", err),
+		)
+		return
+	}
+	if org.JSON200.IsScimEnabled {
+		resp.Diagnostics.AddError(
+			"Invalid Configuration: Cannot create, update or delete a Team resource when SCIM is enabled",
+			"Please disable SCIM in the organization settings to manage Team resources",
+		)
+		return
+	}
+
+	// Validate workspace roles
+	workspaceRoles, diags := common.RequestWorkspaceRoles(ctx, data.WorkspaceRoles)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	for _, role := range workspaceRoles {
+		if !common.ValidateRoleMatchesEntityType(string(role.Role), string(iam.WORKSPACE)) {
+			resp.Diagnostics.AddError(
+				fmt.Sprintf("Role '%s' is not valid for role type '%s'", string(role.Role), string(iam.WORKSPACE)),
+				fmt.Sprintf("Please provide a valid role for the type '%s'", string(iam.WORKSPACE)),
+			)
+			return
+		}
+	}
+
+	// Validate deployment roles
+	deploymentRoles, diags := common.RequestDeploymentRoles(ctx, data.DeploymentRoles)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	for _, role := range deploymentRoles {
+		if !common.ValidateRoleMatchesEntityType(role.Role, string(iam.DEPLOYMENT)) {
+			resp.Diagnostics.AddError(
+				fmt.Sprintf("Role '%s' is not valid for role type '%s'", role.Role, string(iam.DEPLOYMENT)),
+				fmt.Sprintf("Please provide a valid role for the type '%s'", string(iam.DEPLOYMENT)),
+			)
+			return
+		}
```

Comment:
do we have a check where they cannot use same workspaceId or deploymentId for the role
eg. 
workspace_roles = [{workspace_id: cuid1, role: workspace_member}, {workspace_id: cuid1, role: workspace_owner}]

Reply:
no but i can add one